nav2_smac_planner/src/smac_planner_2d.cpp

```diff
@@ -217,6 +217,7 @@ nav_msgs::msg::Path SmacPlanner2D::createPlan(
             std::to_string(start.pose.position.y) + ") was outside bounds");
   }
   _a_star->setStart(mx_start, my_start, 0);
+  _a_star->clearStart(mx_start, my_start);
```

Comment:
Move to `areInputsValid` below the comment, that seems to organize it better. Update for the 3 planners. Adding more external APIs to users of the A* is not ideal, though if there's a good reason its not the end of the world

Reply:
What would you prefer?

Reply:
I'd prefer if clearStart was in `areInputsValid` for Smac